packages/stable/src/api/functions/functionsApi.ts

```diff
@@ -0,0 +1,152 @@
+import {
+  BaseResourceAPI,
+  type CDFHttpClient,
+  type HttpResponse,
+  type MetadataMap,
+} from '@cognite/sdk-core';
+import { FunctionCallsApi } from './functionCallsApi';
+import type {
+  Function as CogniteFunction,
+  FunctionCallRequest,
+  FunctionIdEither,
+  FunctionListResponse,
+  FunctionListScope,
+  FunctionsActivationResponse,
+  FunctionsLimitsResponse,
+  IgnoreUnknownIdsField,
+  LimitList,
+} from './types.gen';
+
+export class FunctionsAPI extends BaseResourceAPI<CogniteFunction> {
+  private readonly functionCallsAPI: FunctionCallsApi;
+
+  constructor(...args: [string, CDFHttpClient, MetadataMap]) {
+    super(...args);
+
+    const [baseUrl, httpClient, map] = args;
+    this.functionCallsAPI = new FunctionCallsApi(baseUrl, httpClient, map);
+  }
+
+  /**
+   * @hidden
+   */
+  public get calls() {
+    return this.functionCallsAPI;
+  }
+
+  /**
+   * [List functions](https://api-docs.cognite.com/20230101/tag/Functions/operation/getFunctions)
+   *
+   * ```js
+   * client.functions.list({ limit: 100 })
+   * ```
+   *
+   */
+  async list(scope?: LimitList): Promise<HttpResponse<FunctionListResponse>> {
+    return await this.post<FunctionListResponse>(this.listPostUrl, {
+      data: scope || { limit: 100 }, // default limit
+    });
+  }
+
+  /**
+   * [Filter functions](https://api-docs.cognite.com/20230101/tag/Functions/operation/listFunctions)
+   *
+   * ```js
+   * client.functions.filter({ filter: { name: 'test' } })
+   * ```
+   *
+   */
+  async filter(
+    scope?: FunctionListScope
+  ): Promise<HttpResponse<FunctionListResponse>> {
+    return await this.post<FunctionListResponse>(this.listPostUrl, {
+      data: scope,
+    });
+  }
+
+  /**
+   * [Retrieve functions](https://api-docs.cognite.com/20230101/tag/Functions/operation/byIdsFunctions)
+   *
+   * ```js
+   * client.functions.retrieve({ items: [{ id: '123' }] })
+   * ```
+   *
+   */
+  async retrieve(
+    scope?: IgnoreUnknownIdsField & { items: FunctionIdEither[] }
+  ): Promise<HttpResponse<FunctionListResponse>> {
+    return await this.post<FunctionListResponse>(this.byIdsUrl, {
+      data: scope,
+    });
```

Comment:
check how assets/timeseries API do it, the default `retrieveEndpoint` gives a bit more (e.g. chunking) 
```suggestion
    return super.retrieveEndpoint(ids, params);
```

Reply:
The API does not support chunking